python/paddle/incubate/nn/functional/build_src_rank_and_local_expert_id.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2025 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import paddle
+from paddle import _C_ops
+
+# from ....framework import LayerHelper, in_dynamic_or_pir_mode
+from paddle.base.framework import in_dynamic_or_pir_mode
+from paddle.base.layer_helper import LayerHelper
+
+if TYPE_CHECKING:
+    from paddle import Tensor
+
+
+def build_src_rank_and_local_expert_id(
+    expert_num_global_tensor: Tensor,
+    expert_num_global: list,
```

Comment:
```suggestion
    expert_num_global: list[Xxx],
```

这里需要补一下泛型内部类型，是 `list[Tensor]` 还是 `list[int]` 之类的

Reply:
好的，收到